src/ray/core_worker/experimental_mutable_object_manager.cc

```diff
@@ -289,16 +299,34 @@ Status MutableObjectManager::ReadAcquire(const ObjectID &object_id,
   // Check whether the channel has an error set before checking that we are the only
   // reader. If the channel is already closed, then it's OK to ReadAcquire and
   // ReadRelease in any order.
+  auto now = std::chrono::steady_clock::now();
+  auto timeout_duration = std::chrono::milliseconds(timeout_ms);
+  auto timeout_point = now + timeout_duration;
+  std::chrono::steady_clock::time_point *timeout_point_ptr = nullptr;
+  if (timeout_ms != -1) {
+    RAY_CHECK_GE(timeout_ms, 0);
+    timeout_point_ptr = &timeout_point;
+  }
+
+  bool locked = false;
+  bool expired = false;
   do {
     RAY_RETURN_NOT_OK(object->header->CheckHasError());
     // The channel is still open. This lock ensures that there is only one reader
     // at a time. The lock is released in `ReadRelease()`.
-  } while (!channel->lock->try_lock());
-  channel->reading = true;
+    locked = channel->lock->try_lock();
+    expired = timeout_point_ptr && std::chrono::steady_clock::now() >= *timeout_point_ptr;
+  } while (!locked && !expired);
+  if (!locked) {
+    // If timeout_ms == 0, we want to try once to get the lock,
+    // therefore we check locked rather than expired.
```

Comment:
Add `CHECK(expired);`

Reply:
hmm, this is already guaranteed by L319?